Sparkle/SUBinaryDeltaApply.m

```diff
@@ -15,38 +15,84 @@
 #include <stdlib.h>
 #include <xar/xar.h>
 
-static void applyBinaryDeltaToFile(xar_t x, xar_file_t file, NSString *sourceFilePath, NSString *destinationFilePath)
+static BOOL applyBinaryDeltaToFile(xar_t x, xar_file_t file, NSString *sourceFilePath, NSString *destinationFilePath)
 {
     NSString *patchFile = temporaryFilename(@"apply-binary-delta");
     xar_extract_tofile(x, file, [patchFile fileSystemRepresentation]);
     const char *argv[] = {"/usr/bin/bspatch", [sourceFilePath fileSystemRepresentation], [destinationFilePath fileSystemRepresentation], [patchFile fileSystemRepresentation]};
-    bspatch(4, (char **)argv);
+    BOOL success = (bspatch(4, (char **)argv) == 0);
     unlink([patchFile fileSystemRepresentation]);
+    return success;
 }
 
 int applyBinaryDelta(NSString *source, NSString *destination, NSString *patchFile)
 {
-    xar_t x = xar_open([patchFile UTF8String], READ);
+    xar_t x = xar_open([patchFile fileSystemRepresentation], READ);
     if (!x) {
-        fprintf(stderr, "Unable to open %s. Giving up.\n", [patchFile UTF8String]);
+        fprintf(stderr, "Unable to open %s. Giving up.\n", [patchFile fileSystemRepresentation]);
         return 1;
     }
+    
+    SUBinaryDeltaMajorVersion majorDiffVersion = FIRST_DELTA_DIFF_MAJOR_VERSION;
+    SUBinaryDeltaMinorVersion minorDiffVersion = FIRST_DELTA_DIFF_MINOR_VERSION;
```

Comment:
> How so? If there's possibility of patch being applied incorrectly, then it should be a major version change.

I should have rephrased this. The patch will apply correctly, but the patch may not have been created correctly. I was thinking of minor versions in cases where there's a bug in creating a patch but not a bug in applying a patch. In this case the major version would not change. A minor version is not needed as you say, unless a human (not the program) wants to know which version of BinaryDelta created the patch.